### DISCUSSION TITLE
How to disable GitHub-hosted runners?

### DISCUSSION THREAD

**Discussion by sl4dy** [community/community#160455] [Actions] [answered]
We can only allow self-hosted runners in our organization.

**Comment by brconnell4** [comment]
I don't think this is the place to ask something like that, that is a better question for github. This is for self-hosted runners. 

Easiest way? 

Set up Branch Protections
Setup up runs on to be self-hosted only
Don't allow commits to master to run

Use reusable workflows that are protected if you want to go even further

**Comment by nebuk89** [comment, marked as answer]
Functionally not yet, we are working on this currently and should have this by the start of next calendar year :)

**Reply by ajith-natarajan-rai** [reply]
Is this available yet?

**Reply by Roshick** [reply]
Any news on this?

**Reply by elee** [reply]
🪶 weight bump on this @nebuk89

**Reply by daniel1941** [reply]
Any update on this?
Can't wait for it!

**Reply by acrogenesis** [reply]
so one year and a half later still not possible?

**Comment by niklasheld** [comment]
We were able to work around this missing feature by specifying SKU-level budgets of $0.00 on every relevant SKU that maps to any of the default GitHub runners. Specifically `actions_linux`, `actions_macos` (seemlingly called „Actions macOS 3-core“ in the UI), `actions_macos_l`, `actions_macos_xl` and `actions_windows`.
It's of course sort of bothersome to configure this number of budgets for all relevant organizations in our enterprise.

Our use-case is to restrict internet-access of GitHub runners by running the GitHub-hosted runners in our Azure VNET.  
In my opinion it is really disappointing and honestly a security issue that this feature is still missing, especially in the light of security incidents such as the Shai-Hulud-malware that used GitHub Actions workflows with access to the internet as means of exfiltration!
@nebuk89, are you by any chance able to share any insights into the GitHub roadmap of this feature, as your last communication here announced this for the start of this year?

Thanks!

**Comment by jackmtpt** [comment]
We were able to get GitHub support to set the hosted runner concurrency to 0 for us via the controls they have in the backend. `Our use-case is to restrict internet-access of GitHub runners by running the GitHub-hosted runners in our Azure VNET.` we wanted to enforce this as well, and now that the internet-connected runners have been forcibly disabled by support we're able to achieve it.
